### PR TITLE
Milestone-02: Blog App - processing data in models

### PR DESCRIPTION
**Pull request by MNisarAli**
### ♻️ Changes Made:

- [x] 🏗️ Created model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
- [x] 📌 Set up associations between "models", with `author_id` column in `posts` table as the `foreign_key` for the `users` table.
- [x] 📊 Familiarized with `rails c` command and used it to insert data into the database.
- [x] 💡 Created custom methods for each model class.
- [x]  🛠️ Tested all methods in `rails c`

### 👀 Impact:

- The project now has properly associated models and data consistency.
- The custom methods provide additional functionality to the project and improve user experience.
- The project is now ready for further development and testing.

### 🙏 Thank you for considering this pull request! 🙏

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  has_many :comments
+  has_many :likes
+
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  def update_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
```
Comment (PinkMoon25) [inline app/models/post.rb:12]: - [ ] [OPTIONAL] Just a suggestion to let you know that we can use [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) *-* `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private

like :   
```
  after_save :update_post_counter

  private

  def update_post_counter
   author.update(posts_counter: author.posts.count)
  end

```

You could do these for all your counters like in likes and comments also :+1:

File: README.md
Comment (PinkMoon25) [inline README.md:1]: - [ ] [OPTIONAl] Now that we are in Rails module how about making the documentation a bit more professional, by adding the necessary steps to run the app locally, currently the getting started section has only setup step mentioned how about adding the install and run step, like `bundle install`, `rails db:create` `rails db:migrate` and `rails server`

**Review by PinkMoon25 — APPROVED** [review]
Hi @MNisarAli , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Rubocop Checks :heavy_check_mark: 
Gitflow :heavy_check_mark: 
data models :heavy_check_mark: 
Assosciations :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (MNisarAli) [reply app/models/post.rb]: Thank you for the suggestion! 👍 I have implemented the Rails Active Callbacks - `after_save` method as you suggested and made the `update_post_counter` method private. I have also used the same approach for other counters like `likes` and `comments` as well. Your feedback has been really helpful! 💯

File: README.md
Comment (MNisarAli) [reply README.md:1]: Hi @PinkMoon25 👋, Thank you for the suggestion! I have improved the [Readme documentation](https://github.com/MNisarAli/Blog-App/pull/2/commits/cc65073207d86961ba9074be6819f3a60a135f80) as you suggested. Your feedback has been really helpful! 💯

**Pull request merged**